
bool is_valid
bool value
